### PR TITLE
Create laravel.yml

### PR DESCRIPTION
**Pull request by DenTray**
# Description

Configure github action with phpunit autotests

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: tests/RemoteDriverTest.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+use RonasIT\Support\AutoDoc\Drivers\RemoteDriver;
+use RonasIT\Support\Tests\Support\Traits\MockTrait;
+use Illuminate\Contracts\Filesystem\FileNotFoundException;
+
+class RemoteDriverTest extends TestCase
+{
+    use MockTrait;
+
+    protected $tmpData;
+    protected $removeDriverClass;
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:14]:
```suggestion
    protected $remoteDriverClass;
```

File: tests/RemoteDriverTest.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    protected $tmpDocumentationFilePath;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->tmpData = $this->getJsonFixture('tmp_data');
+        $this->tmpDocumentationFilePath = __DIR__ . '/../storage/temp_documentation.json';
+
+        $this->removeDriverClass = new RemoteDriver();
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:24]:
```suggestion
        $this->remoteDriverClass = new RemoteDriver();
```

File: tests/support/Traits/MockTrait.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace RonasIT\Support\Tests\Support\Traits;
+
+trait MockTrait
+{
+    protected function mockCLass($className, $methods = [])
```
Comment (astorozhevsky) [inline tests/support/Traits/MockTrait.php:7]:
```suggestion
    protected function mockClass($className, $methods = [])
```

File: tests/RemoteDriverTest.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    public function testSaveTmpData()
+    {
+        $this->removeDriverClass->saveTmpData($this->tmpData);
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:29]:
```suggestion
        $this->remoteDriverClass->saveTmpData($this->tmpData);
```

File: tests/RemoteDriverTest.php [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+        $mock
+            ->expects($this->once())
+            ->method('makeHttpRequest')
+            ->with('post', 'mocked_url/documentations/mocked_key', null, [
+                'Content-Type: application/json'
+            ])
+            ->willReturn(true);
+
+        $mock->saveData();
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:88]: there is no assertion in this test

File: tests/RemoteDriverTest.php [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+        $this->assertFileExists($this->tmpDocumentationFilePath);
+        $this->assertFileEquals($this->generateFixturePath('tmp_data_non_formatted.json'), $this->tmpDocumentationFilePath);
+    }
+
+    public function testGetTmpData()
+    {
+        file_put_contents($this->tmpDocumentationFilePath, json_encode($this->tmpData));
+
+        $result = $this->removeDriverClass->getTmpData();
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:39]:
```suggestion
        $result = $this->remoteDriverClass->getTmpData();
```

File: tests/RemoteDriverTest.php [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+        $this->assertEquals($this->tmpData, $result);
+    }
+
+    public function testGetTmpDataNoFile()
+    {
+        $result = $this->removeDriverClass->getTmpData();
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:46]:
```suggestion
        $result = $this->remoteDriverClass->getTmpData();
```

File: tests/RemoteDriverTest.php [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+        $this->assertNull($result);
+    }
+
+    public function testSaveData()
+    {
+        config(['auto-doc.drivers.remote.key' => 'mocked_key']);
+        config(['auto-doc.drivers.remote.url' => 'mocked_url']);
+
+        $mock = $this->mockCLass(RemoteDriver::class, ['makeHttpRequest']);
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:56]:
```suggestion
        $mock = $this->mockClass(RemoteDriver::class, ['makeHttpRequest']);
```

File: tests/RemoteDriverTest.php [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+        $this->assertFileDoesNotExist($this->tmpDocumentationFilePath);
+    }
+
+    public function testSaveDataWithoutTmpFile()
+    {
+        config(['auto-doc.drivers.remote.key' => 'mocked_key']);
+        config(['auto-doc.drivers.remote.url' => 'mocked_url']);
+
+        $mock = $this->mockCLass(RemoteDriver::class, ['makeHttpRequest']);
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:78]:
```suggestion
        $mock = $this->mockClass(RemoteDriver::class, ['makeHttpRequest']);
```

File: tests/RemoteDriverTest.php [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+        $mock->saveData();
+    }
+
+    public function testGetDocumentation()
+    {
+        config(['auto-doc.drivers.remote.key' => 'mocked_key']);
+        config(['auto-doc.drivers.remote.url' => 'mocked_url']);
+
+        $mock = $this->mockCLass(RemoteDriver::class, ['makeHttpRequest']);
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:96]:
```suggestion
        $mock = $this->mockClass(RemoteDriver::class, ['makeHttpRequest']);
```

File: tests/RemoteDriverTest.php [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    public function testGetDocumentationNoFile()
+    {
+        $this->expectException(FileNotFoundException::class);
+
+        config(['auto-doc.drivers.remote.key' => 'mocked_key']);
+        config(['auto-doc.drivers.remote.url' => 'mocked_url']);
+
+        $mock = $this->mockCLass(RemoteDriver::class, ['makeHttpRequest']);
```
Comment (astorozhevsky) [inline tests/RemoteDriverTest.php:116]:
```suggestion
        $mock = $this->mockClass(RemoteDriver::class, ['makeHttpRequest']);
```

File: tests/support/Traits/SwaggerServiceMockTrait.php [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+use RonasIT\Support\AutoDoc\Drivers\LocalDriver;
+
+trait SwaggerServiceMockTrait
+{
+    use MockTrait;
+
+    protected function mockDriverSaveTmpData($expectedData)
+    {
+        $driver = $this->mockCLass(LocalDriver::class, ['saveTmpData']);
```
Comment (astorozhevsky) [inline tests/support/Traits/SwaggerServiceMockTrait.php:13]:
```suggestion
        $driver = $this->mockClass(LocalDriver::class, ['saveTmpData']);
```

File: tests/SwaggerServiceTest.php [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+        ], [], [
+            'Content-type' => 'application/json'
+        ]);
+
+        $response = new Response($this->getFixture('example_success_roles_response.json'), 200, [
+            'Content-type' => 'application/json',
+            'authorization' => 'Bearer some_token'
+        ]);
+
+        $service->addData($request, $response);
```
Comment (astorozhevsky) [inline tests/SwaggerServiceTest.php:69]: there is not assertion in this test

File: tests/SwaggerServiceTest.php [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+        $request = $this->generateRequest('get', 'users/roles', [
+            'with' => ['users']
+        ]);
+
+        $response = new Response($this->getFixture('example_success_roles_response.json'), 200, [
+            'Content-type' => 'application/json',
+            'authorization' => 'Bearer some_token'
+        ]);
+
+        $service->addData($request, $response);
```
Comment (astorozhevsky) [inline tests/SwaggerServiceTest.php:89]: there is not assertion in this test

File: tests/SwaggerServiceTest.php [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
```
Comment (astorozhevsky) [inline tests/SwaggerServiceTest.php:109]: there is not assertion in this test

File: tests/SwaggerServiceTest.php [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+        ], [
+            'id' => 1,
+            'role-id' => 5
+        ]);
+
+        $response = new Response($this->getFixture('example_success_user_response.json'), 200, [
+            'Content-type' => 'application/json'
+        ]);
+
+        $service->addData($request, $response);
```
Comment (astorozhevsky) [inline tests/SwaggerServiceTest.php:130]: there is not assertion in this test

File: tests/fixtures/StorageDriverTest/tmp_data.json [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+            "example": {
+                "email": "admin@test.com",
+                "password": "123"
+            }
+        }
+    },
+    "info": {
+        "description": "This is automatically collected documentation",
+        "version": "0.0.0",
+        "title": "Photoclaim API",
```
Comment (astorozhevsky) [inline tests/fixtures/StorageDriverTest/tmp_data.json:132]: I think we need to depersonalize the title

File: tests/fixtures/RemoteDriverTest/tmp_data_non_formatted.json [hunk 18]
```diff
@@ -0,0 +1 @@
...eated_at":null,"full_name":"user","settings":null,"new_email":"new_email_test2@test.com","is_email_verified":true,"role":{"id":2,"name":"client","created_at":null,"updated_at":null,"settable":true},"services":[{"id":1,"name":"500px","created_at":"2017-11-16 06:08:34","updated_at":"2017-11-16 06:08:34","is_photo_service":true,"allow_not_unique_accounts":false,"pivot":{"user_id":2,"service_id":1,"is_token_expired":0}}]},"ttl":60,"refresh_ttl":20160}}},"401":{"description":"Unauthorized","schema":{"example":{"error":"You have entered an incorrect credentials."}}},"400":{"description":"Bad Request","schema":{"example":{"error":"The limit of failed authorization attempts has been reached. You can't login in next 50 minutes."}}}},"security":[],"description":"","summary":"login"}}},"definitions":{"authloginObject":{"type":"object","properties":{"email":{"type":"string","description":"2"},"password":{"type":"string","description":""}},"required":["email","password"],"example":{"email":"admin@test.com","password":"123"}}},"info":{"description":"This is automatically collected documentation","version":"0.0.0","title":"Photoclaim API","termsOfService":"","contact":{"email":"your@email.com"}}}
```
Comment (astorozhevsky) [inline tests/fixtures/RemoteDriverTest/tmp_data_non_formatted.json:1]: I think we need to depersonalize the title

File: tests/fixtures/RemoteDriverTest/tmp_data.json [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+            "example": {
+                "email": "admin@test.com",
+                "password": "123"
+            }
+        }
+    },
+    "info": {
+        "description": "This is automatically collected documentation",
+        "version": "0.0.0",
+        "title": "Photoclaim API",
```
Comment (astorozhevsky) [inline tests/fixtures/RemoteDriverTest/tmp_data.json:132]: I think we need to depersonalize the title

File: tests/fixtures/LocalDriverTest/tmp_data_non_formatted.json [hunk 20]
```diff
@@ -0,0 +1 @@
...eated_at":null,"full_name":"user","settings":null,"new_email":"new_email_test2@test.com","is_email_verified":true,"role":{"id":2,"name":"client","created_at":null,"updated_at":null,"settable":true},"services":[{"id":1,"name":"500px","created_at":"2017-11-16 06:08:34","updated_at":"2017-11-16 06:08:34","is_photo_service":true,"allow_not_unique_accounts":false,"pivot":{"user_id":2,"service_id":1,"is_token_expired":0}}]},"ttl":60,"refresh_ttl":20160}}},"401":{"description":"Unauthorized","schema":{"example":{"error":"You have entered an incorrect credentials."}}},"400":{"description":"Bad Request","schema":{"example":{"error":"The limit of failed authorization attempts has been reached. You can't login in next 50 minutes."}}}},"security":[],"description":"","summary":"login"}}},"definitions":{"authloginObject":{"type":"object","properties":{"email":{"type":"string","description":"2"},"password":{"type":"string","description":""}},"required":["email","password"],"example":{"email":"admin@test.com","password":"123"}}},"info":{"description":"This is automatically collected documentation","version":"0.0.0","title":"Photoclaim API","termsOfService":"","contact":{"email":"your@email.com"}}}
```
Comment (astorozhevsky) [inline tests/fixtures/LocalDriverTest/tmp_data_non_formatted.json:1]: I think we need to depersonalize the title

File: tests/fixtures/LocalDriverTest/tmp_data.json [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+            "example": {
+                "email": "admin@test.com",
+                "password": "123"
+            }
+        }
+    },
+    "info": {
+        "description": "This is automatically collected documentation",
+        "version": "0.0.0",
+        "title": "Photoclaim API",
```
Comment (astorozhevsky) [inline tests/fixtures/LocalDriverTest/tmp_data.json:132]: I think we need to depersonalize the title

File: tests/support/Traits/SwaggerServiceMockTrait.php [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+
+        $firstCall = array_merge($expectedData, ['paths' => []]);
+
+        $driver->expects($this->exactly(2))->method('saveTmpData')->withConsecutive([$firstCall], [$expectedData]);
+
+        $this->app->instance(LocalDriver::class, $driver);
```
Comment (astorozhevsky) [inline tests/support/Traits/SwaggerServiceMockTrait.php:19]: are you sure it should be hardcoded as LocalDriver?

**Review by astorozhevsky — CHANGES_REQUESTED** [review]

File: tests/SwaggerServiceTest.php (hunk 14 above)
Comment (DenTray) [reply tests/SwaggerServiceTest.php:89]: `mockDriverSaveTmpData` is also assert that the class methods had been called with expected params

File: tests/SwaggerServiceTest.php (hunk 15 above)
Comment (DenTray) [reply tests/SwaggerServiceTest.php:109]: `mockDriverSaveTmpData` is also assert that the class methods had been called with expected params

File: tests/SwaggerServiceTest.php (hunk 16 above)
Comment (DenTray) [reply tests/SwaggerServiceTest.php:130]: `mockDriverSaveTmpData` is also assert that the class methods had been called with expected params

File: tests/RemoteDriverTest.php (hunk 5 above)
Comment (DenTray) [reply tests/RemoteDriverTest.php:88]: Mock RemoteDriver class also asserting that expected method had been called expected times with expected arguments

File: tests/SwaggerServiceTest.php (hunk 13 above)
Comment (DenTray) [reply tests/SwaggerServiceTest.php:69]: the same as here https://github.com/RonasIT/laravel-swagger/pull/68#discussion_r1068972408

File: tests/RemoteDriverTest.php (hunk 5 above)
Comment (astorozhevsky) [reply tests/RemoteDriverTest.php:88]: mocks and assertions should be separated

File: tests/SwaggerServiceTest.php (hunk 13 above)
Comment (astorozhevsky) [reply tests/SwaggerServiceTest.php:69]: mocks and assertions should be separated

File: tests/SwaggerServiceTest.php (hunk 14 above)
Comment (astorozhevsky) [reply tests/SwaggerServiceTest.php:89]: mocks and assertions should be separated

**Review by astorozhevsky — APPROVED** [review]

**Pull request merged**